shell/src/main/java/alluxio/cli/fs/command/CheckConsistencyCommand.java

```diff
@@ -122,43 +151,85 @@ private void runConsistencyCheck(AlluxioURI path, boolean repairConsistency) thr
       }
     } else {
       Collections.sort(inconsistentUris);
-      System.out.println(path + " has: " + inconsistentUris.size() + " inconsistent files.");
-      List<AlluxioURI> inconsistentDirs = new ArrayList<AlluxioURI>();
+      System.out.println(String.format("%s has: %d inconsistent files. Repairing with %d threads.",
+          path, inconsistentUris.size(), repairThreads));
+      ConcurrentHashSet<AlluxioURI> inconsistentDirs = new ConcurrentHashSet<>();
+
+      ExecutorService svc = Executors.newFixedThreadPool(repairThreads);
+      CompletionService<Boolean> completionService = new ExecutorCompletionService<>(svc);
+      ConcurrentHashSet<Exception> exceptions = new ConcurrentHashSet<>();
+
+      int totalUris = inconsistentUris.size();
       for (AlluxioURI inconsistentUri : inconsistentUris) {
-        URIStatus status = mFileSystem.getStatus(inconsistentUri);
-        if (status.isFolder()) {
-          inconsistentDirs.add(inconsistentUri);
-          continue;
-        }
-        System.out.println("repairing path: " + inconsistentUri);
-        DeletePOptions deleteOptions = DeletePOptions.newBuilder().setAlluxioOnly(true).build();
-        mFileSystem.delete(inconsistentUri, deleteOptions);
-        mFileSystem.exists(inconsistentUri);
-        System.out.println(inconsistentUri + " repaired");
-        System.out.println();
+        completionService.submit(() -> {
+          try {
+            URIStatus status = mFileSystem.getStatus(inconsistentUri);
+            if (status.isFolder()) {
+              inconsistentDirs.add(inconsistentUri);
+              return;
+            }
+            System.out.println("repairing path: " + inconsistentUri);
+            DeletePOptions deleteOptions = DeletePOptions.newBuilder().setAlluxioOnly(true).build();
+            mFileSystem.delete(inconsistentUri, deleteOptions);
+            mFileSystem.exists(inconsistentUri);
+            System.out.println(inconsistentUri + " repaired");
+            System.out.println();
+          } catch (AlluxioException | IOException e) {
+            exceptions.add(e);
+          }
+        }, true);
       }
+
+      waitForTasks(completionService, totalUris, exceptions);
+
+      int totalDirs = inconsistentDirs.size();
       for (AlluxioURI uri : inconsistentDirs) {
```

Comment:
why are dirs repaired separately from files?

Reply:
This was the behavior from previously. I just kept it the same to parallelize everything